### PR TITLE
Add support for pages

### PR DESCRIPTION
**Pull request by akirk**
Followup to #91 

Screenshot 2024-11-07 at 15 54 25

Screenshot 2024-11-07 at 15 54 41

This is just to get this demoable. I was surprised how many changes just adding a single new post type took. I'd really prefer to make this much more modular approach like suggested in https://github.com/WordPress/try-wordpress/discussions/92#discussioncomment-11178162

### CODE REVIEW

**Comment by psrpinto** [comment]
This PR adds the `PagesApi` but with #93 I guess the `PagesApi` needs to be adapted. What do you prefer we do @ashfame @akirk, merge #93 first and then adapt this one, or merge this one and then adapt #93?

File: src/ui/blueprints/blog-post/PageBlueprintEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import { EventTypes } from '@/bus/Event';
+import { ContentEventHandler } from '@/ui/blueprints/ContentEventHandler';
+
+interface Props {
+	blueprint: PageBlueprint;
+	subject: Page;
+	onFieldChanged: ( name: string, field: Field, selector: string ) => void;
+}
+
+export function PageBlueprintEditor( props: Props ) {
```
Comment (psrpinto) [inline src/ui/blueprints/blog-post/PageBlueprintEditor.tsx:16]: Right now this file is under the `/ui/blueprints/blog-post/` directory which was intended just for blog posts, so I think we should either move this file to `/ui/blueprints/page/` or, given that there's in currently only one file per sub directory, move them both to `/ui/blueprints/`.

**Review by psrpinto — COMMENTED** [review]
The UI code looks good to me and I tested locally and it works. The API calls fail but I assume that's expected.

There's a lot of repetition, for example, in `PageBlueprintEditor` and `BlogPostBlueprintEditor` that we should address in the future. We can do so by extracting components that can be used by both, or making a generic `BlueprintEditor`. 

This is related to your comment:

> I was surprised how many changes just adding a single new post type took

Thanks for this feedback, this is something we have been attempting to tackle. We have made attempts to make things more generic (see #65, and also #67 by @ashfame for the API side), but parts of it were later undone because having a completely generic UI and data model that works with any kind of data is not a simple problem to solve.

This effort we have been going through of making it all generic is where the motivation for writing https://github.com/WordPress/try-wordpress/discussions/92 came from. I'm convinced that the ultimate way to solve this problem is to rely on blocks, which I think we should investigate in the future, but for now we can address the repetition tactically, for example by extracting reusable components.

**Comment by ashfame** [comment]
@psrpinto I would be in favor of merging #93 first and then I can take over this PR to make required api endpoint changes to make it work with new endpoints.

**Comment by akirk** [comment]
> but parts of it were later undone because having a completely generic UI and data model that works with any kind of data is not a simple problem to solve.

Could you explain the difficulty here? I feel that you have already come close with what you called [Blueprints](https://github.com/WordPress/try-wordpress/blob/trunk/src/api/Blueprints.ts) (not to mix up with Playground Blueprints!)

In https://github.com/WordPress/try-wordpress/discussions/101 I tried to lay out the data flow in a way that I believe is feasible.

**Comment by psrpinto** [comment]
Ultimately, the difficulty comes from the UI for different kinds of things needing to be different. Specifically navigation, but I think that maybe the solution is to treat navigation differently, at which point there would be no difficulty for making the data model and UI generic for "data post types". I wrote more about this at https://github.com/WordPress/try-wordpress/discussions/101#discussioncomment-11188582

**Comment by ashfame** [comment]
Closing this PR in favor of #100 since that one is merged. But feel free to continue the discussion.

**Pull request closed without merging**